content/blog/2017/10/linux-patching-management-with-ansible.md

```diff
@@ -0,0 +1,524 @@
+        |    |    |
+        |    |    |- group
+        |    |    |    |- <host_group> (optional)
+        |    |    |            |- env_verify.sh
+        |    |    |            |- pre_update.sh
+        |    |    |
+        |    |    |- post_update.sh
+        |    |    |- post_update_conf.env
+        |    |
+        |    |- tasks
+        |         |- main.yaml
+        |
+        |- pre_update
+        |    |
+        |    |- tasks
+        |        |- main.yaml
+        |    
+        |
+        |- post_update
+        |    |
+        |    |- tasks
+        |         |- main.yaml
+        |
+        |- yum_update
+             |
+             |- tasks
+                  |- main.yaml
+
+```
+
+
+### Remote Host Directory Structure
+
+The directory structure on remote hosts is as follows:
+
+```text
+/opt/home/encore/ansibl_svc
+    |
+    |- env_verify.sh
+    |- pre_update.sh
+    |- pre_update.sh
+    |- post_update.sh
+```
+
+
+### Ansible Control Server Configuration
+
+The following examples provide a foundation for getting started with centralized Linux patching for Red Hat Enterprise Linux (RHEL) based systems. Refer to the directory structure wireframed above to aid in visualizing the end result.
+
+
+#### /etc/ansible/inventory/group_vars/all/config
+This configuration file defines the remote destination paths that files are written to by Ansible, and then used during playbook execution to run pre- and post-update actions.  Configurations stored in this file are universal to all remote hosts.  *This configuration file is mandatory.*
+
+```text
+# Destination paths for pre- and post-patching files
+pre_update_script: /home/ansible_svc/pre_update.sh
+post_update_script: /home/ansible_svc/pre_update.sh
+pre_update_script: /home/ansible_svc/pre_update.sh
+pre_update_script: /home/ansible_svc/pre_update.sh
+```
+
+
+#### /etc/ansible/inventory/group_vars/\<host_group\>/config
+This configuration file defines the local source paths for files whose contents are written by Ansible to remote hosts, and then used during playbook execution to run pre- and post-update actions.  Configurations stored in this file are specifc to a host_group.  *This    configuration file is optional.*
+
+**Ex.** */etc/ansible/inventory/group_vars/test_dev/config*
+```yaml
+env_verify_src: "{{ role_path }}/files/group/test_dev/env_verify.sh"
+pre_update_src: "{{ role_path }}/files/group/test_dev/pre_update.sh"
+```
+
+
+#### /etc/ansible/inventory/hosts
+This file contains all the hosts that the Ansible control node will manage.
+```text
+[test_dev] #host_group
+testbuild1.local  #host_record
+```
+
+
+#### /etc/ansible/inventory/host_vars/\<hostname\>/config
+This configuration file defines the local source paths for files whose contents are written by Ansible to remote hosts, and then used during playbook execution to run pre- and post-update actions.  Configurations stored in this file are specific to a host.  *This         configuration file is optional.*
```

Comment:
talk about how this overwrites the data for hostgroups?

also might be good to have a more linear view of the hierarchy?